### PR TITLE
feat: add linting for gamut imports

### PR DESCRIPTION
**Pull request by sylvanasantos**
### Overview

- added a new linting rule to enforce proper path declarations for gamut dependencies

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Testing

1. Checkout `jg-ss-gamut-linting`
2. Run `yarn lint` and ensure everything looks good
3. Add /src to an import statement, rerun `yarn lint`, and make sure you get the correct error
4. Run `yarn lint --fix` and ensure that the import statement gets fixed properly

### CODE REVIEW

File: packages/gamut/temp/lint.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          ) {
+            context.report({
+              messageId: 'includeSrc',
+              node,
+            });
+          }
+          return;
+        }
+
+        if (node.source.value.includes('src')) {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/temp/lint.js:44]: On the off chance a file name happens to include `src`?

```suggestion
        if (node.source.value.includes('/src')) {
```

File: packages/gamut/temp/lint.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          });
+        }
+      },
+    };
+  },
+  defaultOptions: [],
+  meta: {
+    docs: {
+      category: 'Possible Errors',
+      description: 'Ensure module paths are declared correctly',
```
Comment (JoshuaKGoldberg) [inline packages/gamut/temp/lint.js:57]: Could you make this a little more clear? Maybe something about specifically _Gamut_ module paths?

File: packages/gamut/temp/lint.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+module.exports.default = {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/temp/lint.js:1]: > `temp/lint.js`

😛  let's find a better file path than the temporary pairing one. Maybe one of something like...

* `/packages/gamut-eslint/gamut-import-paths`?
* `/packages/gamut-dev-eslint/gamut-import-paths`?
* `/eslint/gamut-import-paths`?

File: packages/gamut/temp/lint.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        if (fileDirectory === importPackage) {
+          context.report({
+            messageId: 'useRelativeImport',
+            node,
+          });
+          return;
+        }
+
+        if (filename.includes('/styleguide/stories')) {
+          if (importPath.includes('/gamut-styles/src')) {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/temp/lint.js:22]: You know, if the path _`endsWith`_ the `/src`(`/`) or `/dist`(`/`) and not just _includes_ it, you could add an auto-fixer to fix it up... what do you think? Up for doing that here, or in a followup?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This looks great! Props for a fantastic first lint rule!

Just requesting changes on the file name being not `temp/lint.js` 😛

**Review by dreamwasp — APPROVED** [review]
this is 😍  😍 

once the` temp/lint.js` file name is changed this is ✨

File: packages/gamut/temp/lint.js (hunk 4 above)
Comment (sylvanasantos) [reply packages/gamut/temp/lint.js]: i think i did this? lol

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.0.12-alpha.3414f4.0<br/>@codecademy/styleguide@27.2.2-alpha.3414f4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602fda4f6be83036e89d9cb7--gamut-preview.netlify.app>https://602fda4f6be83036e89d9cb7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602fda4f6be83036e89d9cb7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.1-alpha.ddd99f.0<br/>@codecademy/styleguide@29.1.1-alpha.ddd99f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602fdc39c0150b3d4f29c713--gamut-preview.netlify.app>https://602fdc39c0150b3d4f29c713--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602fdc39c0150b3d4f29c713>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.1-alpha.a70503.0<br/>@codecademy/styleguide@29.1.1-alpha.a70503.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602fdea3101e66455ee326a4--gamut-preview.netlify.app>https://602fdea3101e66455ee326a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602fdea3101e66455ee326a4>Deploy Logs</a></p>

File: eslint/gamut-import-paths.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        if (fileDirectory === importPackage) {
+          context.report({
+            messageId: 'useRelativeImport',
+            node,
+          });
+          return;
+        }
+
+        if (importPath.includes('/src')) {
+          const indexOfSrc = node.source.range[0] + importPath.search('/src');
```
Comment (JoshuaKGoldberg) [inline eslint/gamut-import-paths.js:22]: This is only used in the `fix` function. You could move it into there. Same with the `indexOfDist` below.

**Review by JoshuaKGoldberg — APPROVED** [review]
This is fantastic!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.11-alpha.7e7136.0<br/>@codecademy/styleguide@29.4.4-alpha.7e7136.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6037af6b55b0191d61698b1b--gamut-preview.netlify.app>https://6037af6b55b0191d61698b1b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6037af6b55b0191d61698b1b>Deploy Logs</a></p>

**Pull request merged**